lib/platform/client/jquery_http.js

```diff
@@ -1,85 +1,101 @@
-                timeout: message.timeout || 0,
-                dataType: "json",
-                success: utils.bind(this, function(data, error, res) {
-                    var response = {
-                        statusCode: res.status,
-                        headers: getHeaders(res.getAllResponseHeaders())
-                    };
-
-                    var complete_response = this._buildResponse(error, response, data);
-                    callback(complete_response);
-                }),
-                error: function(res, data, error) {
-                    var response = {
-                        statusCode: res.status,
-                        headers: getHeaders(res.getAllResponseHeaders())
-                    };
-
-                    if (data === "abort") {
-                        response.statusCode = "abort";
-                        res.responseText = "{}";
-                    }
-                    var json = JSON.parse(res.responseText);
-
-                    var complete_response = that._buildResponse(error, response, json);
-                    callback(complete_response);
-                }
-            };
-            
-            return $.ajax(params);
-        },
-
-        parseJson: function(json) {
-            // JQuery does this for us
-            return json;
-        }
-    });
+
+// Copyright 2011 Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"): you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+(function() {
+    var Http    = require('../../http');
+    var utils   = require('../../utils');
+    var SDK_VERSION = require('../../../package.json').version;
+
+    var root = exports || this;
+
+    var getHeaders = function(headersString) {
+        var headers = {};
+        var headerLines = headersString.split("\n");
+        for(var i = 0; i < headerLines.length; i++) {
+            if (utils.trim(headerLines[i]) !== "") {
+                var headerParts = headerLines[i].split(": ");
+                headers[headerParts[0]] = headerParts[1];
+            }
+        }
+
+        return headers;
+    };
+
+    root.JQueryHttp = Http.extend({
+        init: function(isSplunk) {
+            this._super(isSplunk);
+        },
+
+        makeRequest: function(url, message) {
+            message.headers["Splunk-Client"] = "splunk-sdk-javascript/" + SDK_VERSION;
+            if(message.response_timeout != undefined){
+                message.headers["Response-Timeout"] = message.response_timeout;

```

Comment:
Can you explain a bit about this header? It's not obvious what this is doing, although I see that the cli parses this to set its parameters but I'm not sure why that would be needed since jquery should be the http client that ulimately needs to respect the timeout, I'm not sure why the cli would need this.

Reply:
To keep the code in sync with the backend we had added this header here, but jQuery timeout functionality works differently, so we are thinking of an approach that can be used along with the new abort flow (i.e replacing 600 response status)

Reply:
OK great that sounds good 👍

Reply:
I also noticed since the abort flow has changed we no longer support this advertised feature: https://github.com/splunk/splunk-sdk-javascript/blob/master/CHANGELOG.md#you-can-now-abort-asynchronous-http-requests so I think we need to address this in the docs. For jquery it appears that `abort()` is already supported:
```
let jqxhr = $.ajax( ... );
jqxhr.abort();
```
for needle it looks like the returned object is [advertised as a Readable stream ](https://github.com/tomas/needle#needlemethod-url-data-options-callback--20x) but testing in node it also seems that both `.abort()` and `.destroy()` are supported so maybe we could point users to one or both of those?
```node
let req = needle.get('http://www.google.com', function(error, response) { console.log(`err: ${error} response: ${response}`); });
> req.request.aborted
false
> req.abort();
undefined
> req.request.aborted
true
```
can you confirm? if so that's great that the functionality is still available

Reply:
as per our discussion we have removed the header and now we'll consider the max of `timeout` and `response_timeout` for aborting request.